src/zenml/integrations/vault/secrets_manager/vault_secrets_manager.py

```diff
@@ -0,0 +1,332 @@
+
+        Returns:
+            The secret.
+        """
+        vault_secret_name = self.vault_secret_name(secret_name)
+
+        secret_items = (
+            self.CLIENT.secrets.kv.v2.read_secret_version(
+                path=f"{ZENML_PATH}/{vault_secret_name}",
+                mount_point=self.mount_point,
+            )
+            .get("data", {})
+            .get("data", {})
+        )
+
+        secret_schema = SecretSchemaClassRegistry.get_class(
+            secret_schema=get_secret_schema_name(vault_secret_name)
+        )
+        secret_items["name"] = sanitize_secret_name(secret_name)
+        return secret_schema(**secret_items)
+
+    def vault_list_secrets(self) -> List[str]:
+        """List all secrets in vault without any reformatting.
+
+        This function tries to get all secrets from Vault and returns
+        them as a list of strings all secrets names stored in Vault
+        in the format: <secret_schema>-<secret_name>
+
+        Returns:
+            A list of all secrets in the secrets manager.
+        """
+        self._ensure_client_is_authenticated()
+
+        set_of_secrets: Set[str] = set()
+        try:
+            secrets = self.CLIENT.secrets.kv.v2.list_secrets(
+                path=f"{ZENML_PATH}/", mount_point=self.mount_point
+            )
+        except hvac.exceptions.InvalidPath:
+            logger.debug(
+                f"There are no secrets created within the path `{ZENML_PATH}` "
+            )
+            return list(set_of_secrets)
+
+        secrets_keys = secrets.get("data", {}).get("keys", [])
+        for secret_key in secrets_keys:
+            set_of_secrets.add(secret_key)
+        return list(set_of_secrets)
+
+    def vault_secret_name(self, secret_name: str) -> str:
+        """Get the secret name how it is stored in Vault.
+
+        This function retrieve the secret name in the Vault secrets manager, without
+        any reformatting this secret should be in the format `<secret_schema_name>-<secret_name>`.
+
+        Args:
+            secret_name: The name of the secret to get.
+
+        Returns:
+            The secret name in the Vault secrets manager.
+
+        Raises:
+            KeyError: If the secret does not exist.
+        """
+        self._ensure_client_is_authenticated()
+
+        secrets_keys = self.vault_list_secrets()
+
+        for secret_key in secrets_keys:
+            sanitized_secret_name = sanitize_secret_name(secret_name)
+            secret_name_without_schema = remove_secret_schema_name(secret_key)
+            if sanitized_secret_name == secret_name_without_schema:
+                return secret_key
+        raise KeyError(f"The secret {secret_name} does not exist.")
+
+    def get_all_secret_keys(self) -> List[str]:
+        """Get all secret keys.
+
+        This function tries to get all secrets from Vault and returns
+        them as a list of strings all secrets names without the schema.
```

Comment:
```suggestion
        them as a list of strings. All secrets names are without the schema.
```

Maybe a bit confusing here (and above) to talk about 'secrets names' while in the method it's mentioning secret keys or secret values etc.